sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosContinuablePagedFlux.java

```diff
@@ -0,0 +1,85 @@
+import reactor.core.CoreSubscriber;
+import reactor.core.publisher.Flux;
+
+import java.util.function.Function;
+
+/**
+ * Cosmos implementation of {@link ContinuablePagedFlux}.
+ * <p>
+ * This type is a Flux that provides the ability to operate on pages of type {@link FeedResponse}
+ * and individual items in such pages. This type supports {@link String} type continuation tokens,
+ * allowing for restarting from a previously-retrieved continuation token.
+ * <p>
+ * For more information on the base type, refer {@link ContinuablePagedFlux}
+ *
+ * @param <T> The type of elements in a {@link com.azure.core.util.paging.ContinuablePage}
+ * @see com.azure.core.util.paging.ContinuablePage
+ * @see CosmosPagedFluxOptions
+ * @see FeedResponse
+ */
+public class CosmosContinuablePagedFlux<T> extends ContinuablePagedFlux<String, T, FeedResponse<T>> {
+
+    private final Function<CosmosPagedFluxOptions, Flux<FeedResponse<T>>> optionsFluxFunction;
+
+    CosmosContinuablePagedFlux(Function<CosmosPagedFluxOptions, Flux<FeedResponse<T>>> optionsFluxFunction) {
+        this.optionsFluxFunction = optionsFluxFunction;
+    }
+
+    @Override
+    public Flux<FeedResponse<T>> byPage() {
+        CosmosPagedFluxOptions cosmosPagedFluxOptions = new CosmosPagedFluxOptions();
+
+        return this.optionsFluxFunction.apply(cosmosPagedFluxOptions);
+    }
+
+    @Override
+    public Flux<FeedResponse<T>> byPage(String continuationToken) {
+        CosmosPagedFluxOptions cosmosPagedFluxOptions = new CosmosPagedFluxOptions();
+        cosmosPagedFluxOptions.setRequestContinuation(continuationToken);
+
+        return this.optionsFluxFunction.apply(cosmosPagedFluxOptions);
+    }
+
+    @Override
+    public Flux<FeedResponse<T>> byPage(int preferredPageSize) {
+        CosmosPagedFluxOptions cosmosPagedFluxOptions = new CosmosPagedFluxOptions();
+        cosmosPagedFluxOptions.setMaxItemCount(preferredPageSize);
+
+        return this.optionsFluxFunction.apply(cosmosPagedFluxOptions);
+    }
+
+    @Override
+    public Flux<FeedResponse<T>> byPage(String continuationToken, int preferredPageSize) {
+        CosmosPagedFluxOptions cosmosPagedFluxOptions = new CosmosPagedFluxOptions();
+        cosmosPagedFluxOptions.setRequestContinuation(continuationToken);
+        cosmosPagedFluxOptions.setMaxItemCount(preferredPageSize);
+
+        return this.optionsFluxFunction.apply(cosmosPagedFluxOptions);
+    }
+
+    /**
+     * Subscribe to consume all items of type {@code T} in the sequence respectively.
+     * This is recommended for most common scenarios. This will seamlessly fetch next
+     * page when required and provide with a {@link Flux} of items.
+     *
+     * @param coreSubscriber The subscriber for this {@link CosmosContinuablePagedFlux}
+     */
+    @Override
+    public void subscribe(CoreSubscriber<? super T> coreSubscriber) {
+        Flux<FeedResponse<T>> pagedResponse = this.byPage();
+        pagedResponse.flatMap(tFeedResponse -> {
```

Comment:
each of the reactor transformation will have its own internal buffering, we need to make sure after the transformation, we fetch limited number of items/pages, and possibly configure how many pages should be pre-fetched.

We should have some tests similar to `BackPressureCrossPartitionTest` and `BackpressureTest` on the surface query API.

Not necessarily in this PR. but we need to have that.

Reply:
Sure, will add those. Thanks for the suggestion.

Reply:
@moderakh Added back pressure tests for subscribe call on `CosmosContinuablePagedFlux` object.